libs/shared/src/lib/components/markdown/components/markdown-editor/directives/mardown-editor-selection.directive.ts

```diff
@@ -0,0 +1,57 @@
+import { Directive, ElementRef } from '@angular/core';
+import * as MarkdownIt from 'markdown-it';
+import { MarkdownButton } from '../interfaces';
+
+@Directive({
+    selector: '[markdownEditorSelector]'
+})
+export class MarkdownEditorSelectorDirective {
+    private readonly markdownIt = new MarkdownIt();
+
+    constructor(private readonly hostRef: ElementRef<HTMLTextAreaElement>) {}
+
+    public insertMarkdownSyntax(markdownButton: MarkdownButton): void {
```

Comment:
```suggestion
    public insertMarkdown(markdown: string): void {
```

Давай будемо ззовні отримувати вже markdown готовий і тут його вже якось рендерити. Клас сюди не будемо передавати.

Тобто тут будемо викликати також `this.render` щоб на верхньому рівні про це не думати. Верхній рівень просто передає markdown який треба відрендерити, а тут всі ці умови і процеси нехай будуть всередині